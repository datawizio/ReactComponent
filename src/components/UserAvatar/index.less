@import "~antd/es/avatar/style/index";
@import "../../styles/variables";
