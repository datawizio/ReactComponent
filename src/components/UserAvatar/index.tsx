import React, { useMemo } from "react";
import { Avatar } from "antd";

export interface UserAvatarProps {
  src?: string | null;
  name?: string;
  size?: number;
}

const UserAvatar: React.FC<UserAvatarProps> = ({ src, name, size }) => {
  const abbr = useMemo(
    () =>
      name
        .split(" ")
        .map(i => i[0])
        .join(""),
    [name]
  );

  return (
    <Avatar src={src ? src : undefined} size={size}>
      {abbr}
    </Avatar>
  );
};

export default UserAvatar;
