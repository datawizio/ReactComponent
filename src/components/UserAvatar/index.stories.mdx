import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import UserAvatar from "./index";

<Meta title="Components/UserAvatar" component={UserAvatar} />

# TextArea

Компонент предназначен для вывода аватарки

## Basic

<Preview>
  <Story name="Basic">
    <UserAvatar name="Olloll Trololo" />
  </Story>
</Preview>

## Props

<Props of={UserAvatar} />
