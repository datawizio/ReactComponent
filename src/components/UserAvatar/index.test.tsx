import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import UserAvatar from "./index";

const mockProps = {
  src: "photoSrc",
  name: "John Lennon"
};

const setUp = (props?) => shallow(<UserAvatar {...props} />);

describe("UserAvatar component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("UserAvatar rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("UserAvatar show full name", () => {
    const abbr = mockProps.name
      .split(" ")
      .map(i => i[0])
      .join("");
    expect(toJson(component).children).toContain(abbr);
  });

  it("UserAvatar without photo", () => {
    const wrapper = setUp({ ...mockProps, src: null });

    expect(toJson(component).props).toHaveProperty("src");
    expect(toJson(wrapper).props).not.toHaveProperty("src");
  });
});
