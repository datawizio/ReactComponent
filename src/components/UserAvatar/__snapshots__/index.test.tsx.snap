// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar component UserAvatar rendered correctly 1`] = `
<Avatar
  shape="circle"
  size="default"
  src="photoSrc"
>
  JL
</Avatar>
`;
